apps/api/src/domain/models/user-management/utilities/coscrad-date.entity.ts

```diff
@@ -0,0 +1,19 @@
+import { InternalError } from '../../../../lib/errors/InternalError';
+import { MultilingualText } from '../../../common/entities/multilingual-text';
+import { AggregateCompositeIdentifier } from '../../../types/AggregateCompositeIdentifier';
+import { Aggregate } from '../../aggregate.entity';
+
+export class CoscradDate extends Aggregate {
+    protected validateComplexInvariants(): InternalError[] {
+        throw new Error('Method not implemented.');
+    }
+    getAvailableCommands(): string[] {
+        throw new Error('Method not implemented.');
+    }
+    getName(): MultilingualText {
+        throw new Error('Method not implemented.');
+    }
+    protected getExternalReferences(): AggregateCompositeIdentifier[] {
+        throw new Error('Method not implemented.');
+    }
+}
```

Comment:
You might add a story to add complex invariant validation for a date that ensures that the date is valid given the month and year (considering leap year). Do this on a separate story and don't forget to add a test.
```ts
validateComplexInvariants(): InternalError[]{
    // ...
}
```

Reply:
[Here](https://www.pivotaltracker.com/story/show/187270562) is that story.